### PR TITLE
Docs: Document Reporting API

### PR DESCRIPTION
**Pull request by undef1nd**
**What this PR does / why we need it**:

Adds missing documentation for the reporting API.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/1952

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/developers/http_api/reporting.md [hunk 1]
```diff
@@ -15,10 +15,315 @@ title: Reporting API
...
+| dashboards         | []object  | Dashboards to generate a report for.<br/> [Dashboard object schema](#dashboard-schema).                                                                                                                                                                                                                                                                                                                                                           |
+
+#### Dashboard Schema
+
+| Field name                     | Data type | Description                                              |
+| ------------------------------ | --------- | -------------------------------------------------------- |
+| dashboard.uid                  | string    | Dashboard uid.                                           |
+| timeRange.from                 | string    | Dashboard time range from.                               |
+| timeRange.to                   | string    | Dashboard time range to.                                 |
+| reportVariables.<variableName> | string    | Specifies the name of the dashboard's template variable. |
```
Comment (undef1nd) [inline docs/sources/developers/http_api/reporting.md:303]: @Clarity-89 Could you please take a look at this? Does describing ```[]DashboardDTO `json:"dashboards"``` in this way make sense?

File: docs/sources/developers/http_api/reporting.md [hunk 2]
```diff
@@ -15,10 +15,315 @@ title: Reporting API
...
+| formats            | []string  | Specified what kind of attachment to generate for the report - `csv`, `pdf`, `image`. `pdf` is the default one.                                                                                                                                                                                                                                                                                                                                   |
+| dashboards         | []object  | Dashboards to generate a report for.<br/> [Dashboard object schema](#dashboard-schema).                                                                                                                                                                                                                                                                                                                                                           |
+
+#### Dashboard Schema
```
Comment (Clarity-89) [inline docs/sources/developers/http_api/reporting.md:296]:
```suggestion
#### Report dashboard Schema
```
We call it Report dashboard to differentiate from the actual dashboard data, as this one is specifically for reports. Not sure i this is relevant for the docs though.

File: docs/sources/developers/http_api/reporting.md [hunk 3]
```diff
@@ -15,10 +15,315 @@ title: Reporting API
...
+
+| Field name                     | Data type | Description                                              |
+| ------------------------------ | --------- | -------------------------------------------------------- |
+| dashboard.uid                  | string    | Dashboard uid.                                           |
```
Comment (Clarity-89) [inline docs/sources/developers/http_api/reporting.md:300]:
```suggestion
| dashboard.uid                  | string    | Dashboard UID.                                           |
```
I think the convention is to capitalise these.

File: docs/sources/developers/http_api/reporting.md (hunk 1 above)
Comment (Clarity-89) [inline docs/sources/developers/http_api/reporting.md:303]:
```suggestion
| reportVariables.<variableName> | string    | Key-value pairs containing the template variables for this report, in JSON format. If empty, the template variables from the report's dashboard will be used.  |
```

File: docs/sources/developers/http_api/reporting.md [hunk 4]
```diff
@@ -15,15 +15,446 @@ title: Reporting API
...
+Content-Length: 1840
+
+[
+	{
+		"id": 2,
+		"userId": 1,
+		"orgId": 1,
+		"dashboardId": 0,
+		"dashboardName": "Alerting with TestData",
+		"dashboardUid": "",
```
Comment (Clarity-89) [inline docs/sources/developers/http_api/reporting.md:59]: Maybe we shouldn't include deprecated properties here?

File: docs/sources/developers/http_api/reporting.md [hunk 5]
```diff
@@ -15,15 +15,446 @@ title: Reporting API
...
+	"options": {
+		"orientation": "landscape",
+		"layout": "grid",
+		"timeRange": {
+			"from": "",
+			"to": ""
+		}
+	},
+	"enableDashboardUrl": true,
+	"enableCsv": false,
```
Comment (Clarity-89) [inline docs/sources/developers/http_api/reporting.md:370]: This is deprecated, so probably not necessary here.

File: docs/sources/developers/http_api/reporting.md [hunk 6]
```diff
@@ -71,11 +502,205 @@ Content-Length: 29
...
+| branding.reportLogoUrl   | string    | URL of an image used as a logo on every page of the report.                                                                                                                                                                                                                        |
+| branding.emailLogoUrl    | string    | URL of an image used as a logo in the email.                                                                                                                                                                                                                                       |
+| branding.emailFooterMode | string    | Can be `sent-by` or `none`.<br/>`sent-by` adds a "Sent by `branding.emailFooterText`" footer link to the email. Requires specifying values in `branding.emailFooterText` and `branding.emailFooterLink` fields.<br/>`none` suppresses adding a "Sent by" footer link to the email. |
```
Comment (Clarity-89) [inline docs/sources/developers/http_api/reporting.md:601]:
```suggestion
| branding.emailFooterMode | string    | Can be `sent-by` or `none`.<br/>`sent-by` adds a "Sent by `branding.emailFooterText`" footer link to the email. Requires specifying values in the `branding.emailFooterText` and `branding.emailFooterLink` fields.<br/>`none` suppresses adding a "Sent by" footer link to the email. |
```

**Review by Clarity-89 — COMMENTED** [review]
Looks good, just a suggestion not to include deprecated properties into any sample requests or responses as those properties can be removed at any point.

File: docs/sources/developers/http_api/reporting.md (hunk 4 above)
Comment (undef1nd) [reply docs/sources/developers/http_api/reporting.md:59]: I agree.

**Review by Clarity-89 — APPROVED** [review]

**Pull request merged**